framework/src/meshgenerators/DistributedRectilinearMeshGenerator.C

```diff
@@ -1015,6 +1023,10 @@ DistributedRectilinearMeshGenerator::buildCube(UnstructuredMesh & mesh,
   bool old_allow_renumbering = mesh.allow_renumbering();
   mesh.allow_renumbering(false);
   mesh.allow_find_neighbors(false);
+  // No, I do not want to remove anything in case periodic boundary conditions
+  // are required late. We  stop libMesh from deleting the valuable remote
+  // elements paired with the local elements
+  mesh.allow_remote_element_removal(false);
```

Comment:
@roystgnr doesn't the default libmesh geometric ghosting functor consider periodic boundaries? Perhaps we're just not telling libmesh about periodic boundaries soon enough? Again I would like to get to a point where we can feel comfortable deleting remote elements at any point during mesh construction; to get to that point we need to be confident that we've added the geometric ghosting functors that we need

Reply:
I agreed totally.

 I would like to give minimal information, and `libmesh` takes care of others. The biggest issue is that `libmesh` is able to delete elements, but it can not add elements back. That being said,  I would like to see `libmesh` being able to grow the local subdomain if needed. 

If we can increase the subdomain automatically, then I do not even need to add ghosting elements in a mesh generator.

Based on the current capability we have, the choice I have right now is to stop `libmesh` from deleting anything we added. 

Periodic boundary setup needs to know the mesh information... the whole process is a chicken-egg story.

Reply:
The PeriodicBoundary just needs to know a translation vector or function, and then you just need a boundary_id_type to tell it where it applies.  What mesh information do you need to know?

But yeah - the default ghosting considers periodic boundaries, but you have to have the PeriodicBoundary specifications all attached before you ever reach a prepare_for_use() with allow_remote_element_removal() still set to true.  Setting it to false temporarily in the case where you're already doing distributed generation is probably the best workaround here.

However ... you're already doing distributed generation - you're also manually ghosting potential periodic neighbors, then, I hope?  Relying on MeshCommunication::gather to pull them in wouldn't work, IIRC.

Reply:
Yes, for this special mesh generator `DistributedRectilinearMeshGenerator`, I would like to keep anything I generate, regardless. So I would like to just leave my code as is.

Thanks, everyone!